Examples/Physics_applications/capacitive_discharge/PICMI_inputs_1d.py

```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env python3
+#
+# --- Copyright 2021 Modern Electron
+# --- Monte-Carlo Collision script based on Turner et al. (2013)
+# --- https://doi.org/10.1063/1.4775084
+
+import argparse
+import sys
+
+import numpy as np
+from pywarpx import callbacks, picmi, fields
+from scipy.sparse import csc_matrix
+from scipy.sparse import linalg as sla
+
+constants = picmi.constants
+
+
+class PoissonSolver1D(picmi.ElectrostaticSolver):
```

Comment:
Is this solver needed? There is a 1D Poisson solver already in the C++. If this is needed, could a comment be added explaining why this is used and not the C++ solver?

Reply:
I wasn't aware that there was a direct solver implemented on the C++ side. Where can I find that? Also would it be accessible from Python?

Reply:
Also it might be good to leave the python solver in here regardless since it also serves as a test of `fields.py` (write) functionality for a 1D simulation.

Reply:
I'm referring to the 1D solver that is part of `WarpX::computePhi`. It is not a direct solver though. It is fine to keep the Python solver here, but it would be helpful to have a comment about why it is needed.

Reply:
Oh I see. Sounds good; I will add a comment about why it is used. It is just because in the simple 1d system a direct solver is much faster than a MLMG one.

Reply:
I know for 2D with a long, skinny grid, that the direct solver is faster than the iterative. Though interesting that the iterative solver would be much slower in 1D.

Reply:
Actually I just assumed it would be but in a quick test (running this input script to 10000 steps on 2 procs) the AMReX MLMG solver is actually faster than the direct solver. The direct solver case took on average ~8 ms per step while the MLMG solver took only ~6 ms per step.

Reply:
In light of the above I will go ahead and remove the direct solver. We will still have partial coverage of `fields.py` functionality in 1d since the ion charge density gathering will use it, we just won't have coverage of writing to fields.